src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -969,7 +972,17 @@ export class TreeModel {
 		let eventData: IRefreshEvent = { item: item, recursive: recursive };
 		this._onRefresh.fire(eventData);
 		return item.refresh(recursive).then(() => {
```

Comment:
While you're here let's make this async to simplify it

Reply:
Do you mean the inner function or the refresh function itself?